code/datums/diseases/critical.dm

```diff
@@ -133,4 +138,65 @@
 					affected_mob.emote(pick("twitch", "gasp"))
 				if(prob(5) && ishuman(affected_mob))
 					var/mob/living/carbon/human/H = affected_mob
-					H.set_heartattack(TRUE)
\ No newline at end of file
+					H.set_heartattack(TRUE)
+
+/datum/disease/critical/hypoglycemia
+	name = "Hypoglycemia"
+	form = "Medical Emergency"
+	max_stages = 3
+	spread_flags = SPECIAL
+	spread_text = "The patient has low blood sugar."
+	cure_text = "Eating or glucose treatment"
+	viable_mobtypes = list(/mob/living/carbon/human)
+	stage_prob = 1
+	severity = DANGEROUS
+	disease_flags = CURABLE
+	bypasses_immunity = TRUE
+	virus_heal_resistant = TRUE
+
+/datum/disease/critical/hypoglycemia/has_cure()
+	if(ishuman(affected_mob))
+		var/mob/living/carbon/human/H = affected_mob
+		if(NO_HUNGER in H.dna.species.species_traits)
+			return TRUE
+		if(ismachine(H))
+			return TRUE
+	return ..()
+
+/datum/disease/critical/hypoglycemia/stage_act()
+	if(..())
+		if(affected_mob.nutrition > NUTRITION_LEVEL_HYPOGLYCEMIA)
+			to_chat(affected_mob, "<span class='notice'>You feel a lot better!</span>")
+			cure()
+			return
+		switch(stage)
+			if(1)
+				if(prob(4))
+					to_chat(affected_mob, "<span class='warning'>You feel hungry!</span>")
+				if(prob(2))
+					to_chat(affected_mob, "<span class='warning'>You have a headache!</span>")
+				if(prob(2))
+					to_chat(affected_mob, "<span class='warning'>You feel [pick("anxious", "depressed")]!</span>")
+			if(2)
+				if(prob(4))
+					to_chat(affected_mob, "<span class='warning'>You feel like everything is wrong with your life!</span>")
+				if(prob(5))
+					affected_mob.Slowed(rand(4, 16))
+					to_chat(affected_mob, "<span class='warning'>You feel [pick("tired", "exhausted", "sluggish")].</span>")
+				if(prob(5))
+					affected_mob.Weaken(6)
+					affected_mob.Stuttering(10)
+					to_chat(affected_mob, "<span class='warning'>You feel [pick("numb", "confused", "dizzy", "lightheaded")].</span>")
+					affected_mob.emote("collapse")
+			if(3)
+				if(prob(8))
+					var/datum/disease/D = new /datum/disease/critical/shock
```

Comment:
Shock? For not eating?
Can't shock advance to other, lethal conditions?
It seems really weird that someone could fall into shock, then heart failure, then die, just because they did not eat for two hours.
Even if you assume a typical work shift is more like 8 hours, compressed.... people skip lunch all the time. It doesn't kill them.
This seems over the top.

Reply:
Doesn't shock heal itself if your health is high enough anyway? Am I misremembering newcrit?

Reply:
Yeah it has a chance to heal itself if you have high health

Reply:
It's part of the PR as well. It will heal if you have over 25 HP currently. The PR adds the condition that you need over the threshold amount of hunger. So this just means don't be starving and you'll heal from it. Unless you're dying